helsingborg-stad/app-mitt-helsingborg
# Feat: Dynamically set app in dev mode

## Explain the changes you’ve made

Created a solution for entering the dev environment even when the app is build for production.
Also moved dev features to a new screen that is shown only in "dev mode".

## Explain why these changes are made

This is needed by Apple and Google when they are reviewing the app.  

## Explain your solution

Added a new .env variable that defines a test personal number. If that specific personal number is entered when logging in the app is redirecting all requests from prod(AWS) to dev. You are still required to log in with Bankid but will enter the test-environment instead. 
To change and access the dev/prod mode dynamically I created a new context for this.

Because context(and other hooks) cannot be accessed in helper functions, I also needed  to save the "app env" value in AsyncStorage for easy accessing. This part feels is a little hacky but was the best I could come up with, any suggestions for other solutions is welcome. 

Created a new screen for dev features that is shown in dev mode and can be accessed from the Profile screen.

## How to test the changes?

1. Add these .env variables:
APP_ENV=production
MITTHELSINGBORG_IO=[prod or stage aws url]
MITTHELSINGBORG_IO_APIKEY=[some api key]
MITTHELSINGBORG_IO_DEV=[develop aws url]
MITTHELSINGBORG_IO_DEV_APIKEY=[some api key]
TEST_PERSONAL_NUMBER=[199803312389] (person you have in test-Bankid)
USE_BANKID=true

2. Rebuild the app
3. Try to log in with a personal number that is not your test-person. And the app should request your prod-url.
4. To test dev environment. Log in with your test-user and the app should request your dev-url.
5. The app is now in "dev mode" and if you click on Profile screen you should see a button with "Utvecklarfunktioner".


## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.
